### PR TITLE
B 19 add filters for pages by role

### PR DESCRIPTION
**Pull request by Rooman**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -75,11 +75,7 @@
             <artifactId>thymeleaf</artifactId>
             <version>3.0.11.RELEASE</version>
         </dependency>
-        <dependency>
-            <groupId>org.thymeleaf.extras</groupId>
-            <artifactId>thymeleaf-extras-java8time</artifactId>
-            <version>3.0.4.RELEASE</version>
-        </dependency>
+
```
Comment (Brutt) [inline pom.xml]: exclude pom.xml from PR

File: src/main/java/com/photostudio/web/filter/SecurityFilter.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.photostudio.entity.user.UserRole.*;
+
+
+@Slf4j
```
Comment (Brutt) [inline src/main/java/com/photostudio/web/filter/SecurityFilter.java:23]: Use approach as implemented in other classes

File: src/main/java/com/photostudio/web/filter/SecurityFilter.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        String excludedUrlsStr = filterConfig.getInitParameter("excludedUrls");
+        this.excludedUrls = Arrays.asList(excludedUrlsStr.split(","));
+
+        urlToRoleMap.put("/logout", USER);
+        urlToRoleMap.put("/orders", USER);
+        // Optional, included by default
+        urlToRoleMap.put("/admin/order", ADMIN);
```
Comment (Brutt) [inline src/main/java/com/photostudio/web/filter/SecurityFilter.java]: we don`t have this endpoint

File: src/main/java/com/photostudio/web/filter/SecurityFilter.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            log.debug("Access to excluded url {}", servletPath);
+        } else if (session == null) {
+            response.sendRedirect(request.getContextPath() + "/login");
+        } else {
+            UserRole requiredRole = urlToRoleMap.getOrDefault(servletPath, ADMIN);
+            User currentUser = session.getUser();
+            if (hasAccess(requiredRole, currentUser.getUserRole())) {
+                chain.doFilter(request, response);
+            } else {
+                response.sendRedirect(request.getContextPath() + "/fuck_you_asshole");
```
Comment (Brutt) [inline src/main/java/com/photostudio/web/filter/SecurityFilter.java]: change endpoint name

File: src/main/java/com/photostudio/web/templater/TemplateEngineFactory.java [hunk 5]
```diff
@@ -33,7 +31,7 @@ public static void configTemplate(ServletContext servletContext) {
 
     public static void process(String template, Map<String, Object> productsMap, Writer writer) {
         IContext context = new Context(Locale.getDefault(), productsMap);
-        TEMPLATE_ENGINE.addDialect(new Java8TimeDialect());
```
Comment (Brutt) [inline src/main/java/com/photostudio/web/templater/TemplateEngineFactory.java]: do not overwrite develop`s branch changes

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -1,5 +1,5 @@
 !Connect to the DB MySQL:
 jdbc.url=jdbc:mysql://localhost:3306/photostudio
 jdbc.user=root
-jdbc.password=toor
+jdbc.password=root
```
Comment (Yuliia-Sh) [inline src/main/resources/application.properties]: password for server will  be incorrect, won't it?

File: src/main/java/com/photostudio/web/filter/SecurityFilter.java (hunk 2 above)
Comment (Rooman) [reply src/main/java/com/photostudio/web/filter/SecurityFilter.java:23]: Не менять.

File: src/main/resources/application.properties (hunk 6 above)
Comment (Brutt) [reply src/main/resources/application.properties]: Server doesn't use credentials from property file. It uses credentials from system variables

**Review by ValentynaBondarenko — APPROVED** [review]

**Review by Yuliia-Sh — APPROVED** [review]

**Review by Brutt — APPROVED** [review]

**Review by shaguahill — APPROVED** [review]

**Pull request merged**